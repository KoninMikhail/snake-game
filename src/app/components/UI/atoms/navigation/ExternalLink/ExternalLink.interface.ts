//
// InternallLink
// ----------------------------------------------------------------------

export interface IExternalLinkProps {
    href: string;
    target?: string;
    rel?: string;
}
