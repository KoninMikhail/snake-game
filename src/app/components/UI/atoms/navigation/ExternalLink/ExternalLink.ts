import styled from 'styled-components';
import {IExternalLinkProps} from '@ui/atoms/navigation/ExternalLink/ExternalLink.interface';

/**
 * @name ExternalLinkProps
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for external links
 *
 * @param href
 * @param target
 * @param rel
 * @returns {JSX.Element}
 */
export const ExternalLink = styled.a.attrs<IExternalLinkProps>((props) => ({
    href: props.href ?? 'https://google.com',
    target: props.target ?? '_blank',
    rel: props.rel ?? 'noopener noreferrer',
}))`
    cursor: pointer;
    text-decoration: none;
    color: white;

    &:hover {
        text-decoration: underline;
    }
`;
