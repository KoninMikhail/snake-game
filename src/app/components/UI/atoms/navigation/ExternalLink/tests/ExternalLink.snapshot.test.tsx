import { render } from '@testing-library/react';
import { ExternalLink } from '../ExternalLink';

/**
 * Tests
 */
describe('ExternalLink', () => {
    it('equal snapshot', () => {
        const { container } = render(
            <ExternalLink href="https://www.google.com/" target="_self" rel="nofollow">
                {'yeah!'}
            </ExternalLink>
        );

        expect(container).toMatchSnapshot();
    });
});
