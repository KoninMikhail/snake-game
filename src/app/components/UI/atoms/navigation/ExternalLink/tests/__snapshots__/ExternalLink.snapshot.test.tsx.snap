// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink equal snapshot 1`] = `
.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <a
    class="c0"
    href="https://www.google.com/"
    rel="nofollow"
    target="_self"
  >
    yeah!
  </a>
</div>
`;
